src/HealthChecks/HealthChecks/test/DefaultHealthCheckServiceTest.cs

```diff
@@ -20,36 +20,6 @@ namespace Microsoft.Extensions.Diagnostics.HealthChecks;
 
 public class DefaultHealthCheckServiceTest
 {
-    [Fact]
-    public void Constructor_ThrowsUsefulExceptionForDuplicateNames()
-    {
-        // Arrange
-        //
-        // Doing this the old fashioned way so we can verify that the exception comes
-        // from the constructor.
-        var serviceCollection = new ServiceCollection();
-        serviceCollection.AddLogging();
-        serviceCollection.AddOptions();
-        serviceCollection.AddHealthChecks()
-            .AddCheck("Foo", new DelegateHealthCheck(_ => Task.FromResult(HealthCheckResult.Healthy())))
-            .AddCheck("Foo", new DelegateHealthCheck(_ => Task.FromResult(HealthCheckResult.Healthy())))
-            .AddCheck("Bar", new DelegateHealthCheck(_ => Task.FromResult(HealthCheckResult.Healthy())))
-            .AddCheck("Baz", new DelegateHealthCheck(_ => Task.FromResult(HealthCheckResult.Healthy())))
-            .AddCheck("Baz", new DelegateHealthCheck(_ => Task.FromResult(HealthCheckResult.Healthy())));
-
-        var services = serviceCollection.BuildServiceProvider();
-
-        var scopeFactory = services.GetRequiredService<IServiceScopeFactory>();
-        var options = services.GetRequiredService<IOptions<HealthCheckServiceOptions>>();
-        var logger = services.GetRequiredService<ILogger<DefaultHealthCheckService>>();
-
-        // Act
-        var exception = Assert.Throws<ArgumentException>(() => new DefaultHealthCheckService(scopeFactory, options, logger));
```

Comment:
Let's add this test back. It might be better to make this line call `services.GetRequiredService<HealthCheckService>()` and verify we get the same exception since that's public API.

Reply:
Restored to its original form - didn't have any call to `services.GetRequiredService<HealthCheckService>()`, instead a DefaultHealthCheckService was created via ctor.